fast_llm/models/gpt/model.py

```diff
@@ -35,24 +34,43 @@ def __init__(
         config: GPTBaseModelConfig,
         distributed_config: DistributedConfig,
     ):
-        self._hidden_dim = TensorDim("hidden", config.embeddings_layer.hidden_size)
         super().__init__(config, distributed_config)
+
+        self._hidden_dim = TensorDim("hidden", config.embeddings.hidden_size)
+        self.embeddings: LanguageModelEmbedding = self._config.embeddings.get_layer(
+            distributed_config,
+            hidden_dim=self._hidden_dim,
+            lr_scale=None,
+            peft=self._config.peft,
+        )
+        self.decoder = self._config.decoder.get_layer(
+            distributed_config,
+            self._hidden_dim,
+            lr_scale=None,
+            peft=self._config.peft,
+        )
+        self.head = self._config.head.get_layer(
+            distributed_config,
+            self._config.embeddings,
+            hidden_dim=self._hidden_dim,
+            lr_scale=None,
+            peft=self._config.peft,
+        )
+
         if self._config.use_megatron_initialization:
             for param in self.parameters():
                 Assert.custom(isinstance, param, ParameterMeta)
                 param.init_parameter = get_init_megatron(
-                    param, self._config.decoder.block, config.embeddings_layer.hidden_size
+                    param, self._config.decoder.block, config.embeddings.hidden_size
                 )  # Noqa
-        # `self._reference_models` is not populated at this point, so we pass a mutable dict.
-        self._preprocessors: list[Preprocessor] = self._config.get_preprocessors(distributed_config)
 
-    def get_layers(self) -> list[Layer]:
-        return self._config.get_blocks(self._distributed_config)
+    def get_layers(self) -> list["Layer"]:
+        return self.embeddings.get_layers() + self.decoder.get_layers() + self.head.get_layers()
 
     def preprocess_meta(
```

Comment:
remove?

Reply:
Not possible atm, it's no longer in the preprocessor/layer interface but is still there for base models. Planning to revisit this with the batch-splitting mechanism, but leaving for later.